### DISCUSSION TITLE
Hardening OS takes long time to complete

### DISCUSSION THREAD

**Discussion by Naranthiran** [ComplianceAsCode/content#10323] [General]
Hi,

Hardening the OS is taking more time when compared to the older versions of scap-security-guide.

We are executing the below script which takes more than 20 minutes to complete.

/usr/share/scap-security-guide/bash/almalinux8-script-stig.sh

SCAP Security Guide Version:
scap-security-guide-0.1.60-7.el8.alma.noarch

Operating System Version:
AlmaLinux 8.6 (Sky Tiger)

Steps to Reproduce:
Install AlmaLinux-8.6
install scap-security-guide-0.1.60-7.el8.alma.noarch
3.chmod +x /usr/share/scap-security-guide/bash/almalinux8-script-stig.sh
./usr/share/scap-security-guide/bash/almalinux8-script-stig.sh
Actual Results:
Expected Results:
Wanted to fasten the hardening time

Additional Information/Debugging Steps:
Regards
Naranthiran Duraisamy

**Comment by marcusburghardt** [comment]
The time to execute a profile remediation depends on different factors, like the system resources, current state of the system, number of selected rules, type of remediation, etc.

If you notice any specific rule or task that you consider to be taking much longer time than expected, please file a Upstream Issue with details so we can investigate it.

For reference, in general according to my tests, it takes less than 5 minutes in RHEL8.
